nickel/src/capi.rs

```diff
@@ -0,0 +1,1016 @@
+//! C bindings for the Nickel language
+//!
+//! TODO: example
+
+#![allow(clippy::missing_safety_doc)]
+#![allow(non_camel_case_types)]
+#![deny(missing_docs)]
+
+use std::{
+    ffi::{c_char, c_int, c_void, CStr},
+    io::Write,
+};
+
+use nickel_lang_core::term::Term;
+
+use crate::{Array, Context, Error, ErrorFormat, Expr, Number, Record, Trace, VirtualMachine};
+
+/// The main entry point.
+pub struct nickel_context {}
+
+impl nickel_context {
+    unsafe fn as_rust_mut(this: &mut *mut Self) -> &mut Context {
```

Comment:
Even if it's a bit of copy-pasting and sounds trivial, I think any non-`extern "C"` unsafe function should be documented with the safety pre-conditions, as we would do for any `unsafe` block, whether it is exposed or not.